src/components/PopupWithForm.js

```diff
@@ -7,46 +7,44 @@ export default class PopupWithForm extends Popup {
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
     this._inputSelector = config.inputSelector;
-    this._submitButtonSelector = config.submitButtonSelector;
-    //this._submitButton = this._element.querySelector(".modal__button");
+
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+
+    this._defaultButtonText = this._submitButton.textContent = "Save";
   }
 
   open() {
     super.open();
-    this._popupForm.addEventListener("submit", this._handleFormSubmit);
+    this._popupForm.addEventListener("submit", this._handleFormSubmitMethod);
   }
 
   close() {
-    console.log("FOCUSED CLOSE ON");
     super.close();
-    this._popupForm.removeEventListener("submit", this._handleFormSubmit);
+    this._popupForm.removeEventListener("submit", this._handleFormSubmitMethod);
     this._popupForm.reset();
   }
+
   _getInputValues() {
     const inputValues = {};
 
     this._popupForm.querySelectorAll(this._inputSelector).forEach((input) => {
       inputValues[input.name] = input.value;
     });
-
     return inputValues;
   }
 
-  _handleFormSubmit(event) {
+  submitButtonLoadingState(isLoading) {
+    if (this._submitButton) {
+      this._submitButton.textContent = isLoading
+        ? this._defaultButtonText
+        : "Saving...";
+    }
+  }
+
+  _handleFormSubmitMethod = (event) => {
     event.preventDefault();
     const inputValues = this._getInputValues();
     this._handleFormSubmit(inputValues);
     this.close();
```

Comment:
Please, delete `this.close();` from line 48 in `PopupWithForm` 

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**